src/main/java/com/google/devtools/build/lib/syntax/UnaryOperatorExpression.java

```diff
@@ -63,30 +63,33 @@ private static Object evaluate(TokenKind op, Object value, Location loc)
         return !EvalUtils.toBoolean(value);
 
       case MINUS:
-        if (!(value instanceof Integer)) {
-          throw new EvalException(
-              loc,
-              String.format(
-                  "unsupported operand type for -: '%s'", EvalUtils.getDataTypeName(value)));
-        }
-        try {
-          return Math.negateExact((Integer) value);
-        } catch (ArithmeticException e) {
-          // Fails for -MIN_INT.
-          throw new EvalException(loc, e.getMessage());
+        if (value instanceof Integer) {
+          try {
+            return Math.negateExact((Integer) value);
+          } catch (ArithmeticException e) {
+            // Fails for -MIN_INT.
+            throw new EvalException(loc, e.getMessage());
+          }
         }
+        break;
+
       case PLUS:
-        if (!(value instanceof Integer)) {
-          throw new EvalException(
-              loc,
-              String.format(
-                  "unsupported operand type for +: '%s'", EvalUtils.getDataTypeName(value)));
+        if (value instanceof Integer) {
+          return value;
         }
-        return value;
+        break;
 
-      default:
```

Comment:
Please restore the default case (it can be empty in this case). Some static analyzers report errors if not all enum values are handled, adding `default:` with an explanation in a comment will fix that (and also make it clear that there are other unhandled cases).